includes/discovery/fdb-table/iosxe.inc.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+$vtpdomains = snmpwalk_cache_oid($device, 'vlanManagementDomains', array(), 'CISCO-VTP-MIB');
+$vlans      = snmpwalk_cache_twopart_oid($device, 'vtpVlanEntry', array(), 'CISCO-VTP-MIB');
+
+foreach ($vtpdomains as $vtpdomain_id => $vtpdomain) {
+    echo "VTP Domain $vtpdomain_id {$vtpdomain['managementDomainName']}> ";
+    foreach ($vlans[$vtpdomain_id] as $vlan_raw => $vlan) {
+        echo "$vlan_raw ";
+        if (!array_key_exists($vlan_raw, $vlans_dict)) {
+            $newvlan_id = dbInsert(array(
+                'device_id' => $device['device_id'],
+                'vlan_domain' => $vtpdomain_id,
+                'vlan_vlan' => $vlan_raw,
+                'vlan_name' => $vlan['vtpVlanName'],
+                'vlan_type' => $vlan['vtpVlanType']
+            ), 'vlans');
+            $vlans_dict[$vlan_raw] = $newvlan_id;
+        }
+
+        if ($vlan['vtpVlanState'] == 'operational') {
+            $device_vlan = array_merge($device, array('community' => $device['community'] . '@' . $vlan_raw, 'context_name' => "vlan-$vlan_raw"));
+
+            $FdbPort_table = snmp_walk($device_vlan, 'dot1dTpFdbPort', '-OqsX', 'BRIDGE-MIB');
```

Comment:
If you change the `snmp_walk` calls to `snmpwalk_cache_*` calls then you don't need to explode on new lines.